test/test_kernel.cpp

```diff
@@ -0,0 +1,216 @@
+//Here, the sources of kernels are tested.
+#include "yacx/Headers.hpp"
+#include "yacx/KernelArgs.hpp"
+#include "yacx/Kernel.hpp"
+#include "yacx/Source.hpp"
+
+#include <catch2/catch.hpp>
+#include <iostream>
+
+using yacx::Kernel, yacx::KernelTime, yacx::KernelArg, yacx::Source, 
+yacx::Header, yacx::Headers;
+
+TEST_CASE("The kernel - source code will be tested under the following conditions."){
+    //A. Preparing the input for the kernel-compilation using source
+    int datasize{10};
+    int *hX = new int[10]{1,2,3,4,5,6,7,8,9,10};
+    int *hY = new int[10]{6,7,8,9,10,11,12,13,14,15};
+    int *hOut_1 = new int[10]{7,9,11,13,15,1,2,3,4,5};
+    int *hOut_2 = new int[10]{7,9,11,13,15,1,2,3,4,5};
+    int *hOut_3 = new int[10]{7,9,11,13,15,1,2,3,4,5};
+    int *hOut_4 = new int[10]{7,9,11,13,15,1,2,3,4,5};
+
+    size_t bufferSize = datasize*sizeof(int);
+    
+    std::vector<KernelArg> args;
+    args.emplace_back(KernelArg{hX, bufferSize});
+    args.emplace_back(KernelArg{hY, bufferSize});
+    args.emplace_back(KernelArg{hOut_1, bufferSize, true});
+    args.emplace_back(KernelArg(&datasize));
+
+    Headers headers;
+    headers.insert(Header{"cuda_runtime.h"});
+
+    //A2. Preparing the output for kernel-compilation
+    int *hostCompareOutput = new int[10]{7,9,11,13,15,17,19,21,23,25};
+    
+    //B1. A Kernel must be created in a controlled manner.
+    SECTION("1. A Kernel must be created in a controlled manner."){
+        Source source{"#include \"cuda_runtime.h\"\n"
+        "extern \"C\"\n"
+        "__global__ void cuda_add(int *x, int *y, int *out, int "
+        "datasize) {\n"
+        " int i = threadIdx.x;\n"
+        " out[i] = x[i] + y[i];\n"
+        "}", headers};
+
+        dim3 grid_test(1);
+        dim3 block_test(10);
+
+        //B1A1. Controlled results from controlled inputs.
+        REQUIRE_NOTHROW(source.program("cuda_add").compile().configure(grid_test, block_test).launch(args));
+        
+        for (int i=0;i<10;i++){
+            REQUIRE(hOut_1[i] == hostCompareOutput[i]);
+        }
+    }
+
+    /*B2. Launching a kernel using the following block and grid dimensions as well as
+    naming conventions for the kernel-function.*/
+    //B2A. Checking for consistencies in the naming convention of kernel function.
+    SECTION("2A. The created kernel must be launched with proper kernel-function names."){
+        Source source{"#include \"cuda_runtime.h\"\n"
+            "extern \"C\"\n"
+            "__global__ void cuda_add(int *x, int *y, int *out, int "
+            "datasize) {\n"
+            " int i = threadIdx.x;\n"
+            " out[i] = x[i] + y[i];\n"
+            "}", headers};
+
+        dim3 grid_test(1);
+        dim3 block_test(10);
+
+        REQUIRE_THROWS(source.program("cuda_addition").compile().configure(grid_test, block_test).launch(args));
+        REQUIRE_NOTHROW(source.program("cuda_add").compile().configure(grid_test, block_test).launch(args));
```

Comment:
these are the exact tests I want for Source, maybe just cut them here and paste them there ^^